Accord journal purger

ifesdjeen opened the pull request.

belliottsmith committed:
CASSANDRA-19967: Refine epoch quorum intersections, sync points, conflict relations and outcome inference

- ExclusiveSyncPoints must imply global cluster properties, by executing on all participating epochs so that across all healthy replicas we can know all lower TxnId are durably applied
- Conflict relations are managed at Token rather than Key level to improve memory occupancy for workloads with large keys
- Epoch quorum intersections have been updated: preaccept no longer needs to chase the execution epoch, and recovery now contacts unmigrated epochs for dependency calculation; fast path decisions only involve the proposal epoch. This behaviour is now partially configurable.
- Simplifies and fixes the inference of a transaction's outcome when some shards have truncated the transaction
- Refined internal representations of key ownership across epoch changes

iamaleksey commented on src/java/org/apache/cassandra/service/accord/AccordService.java:
```diff
@@ -1073,4 +1079,9 @@ public CompactionInfo getCompactionInfo()
         return new CompactionInfo(redundantBefores, ranges, durableBefore.get());
     }
+
+    public Params journalConfiguration()
```
Nit: missing `@Override`

iamaleksey commented on src/java/org/apache/cassandra/service/accord/AccordService.java:
```diff
@@ -271,6 +272,11 @@ public CompactionInfo getCompactionInfo()
             return new CompactionInfo(new Int2ObjectHashMap<>(), new Int2ObjectHashMap<>(), DurableBefore.EMPTY);
         }
 
+        public Params journalConfiguration()
```
Nit: missing `@Override`

iamaleksey commented on src/java/org/apache/cassandra/db/compaction/CompactionIterator.java:
```diff
@@ -703,298 +661,6 @@ protected UnfilteredRowIterator applyToPartition(UnfilteredRowIterator partition
-                    // Eventually the tombstone can be dropped by `durableBefore.min(txnId) == Universal`
-                    // We can still encounter sliced command state just because compaction inputs are random
-                    return BTreeRow.emptyDeletedRow(row.clustering(), new Row.Deletion(DeletionTime.build(row.primaryKeyLivenessInfo().timestamp(), nowInSec), false));
-
```
I think these two lines got lost in transition. Bad merge?

ifesdjeen committed:
Add purging to Accord Journal table

Patch by Alex Petrov; reviewed by Aleksey Yeshchenko and Benedict Elliott Smith for CASSANDRA-19877
